test/e2e/autotls/auto_tls_test.go

```
@@ -22,92 +20,110 @@ import (
 
-	cancel = turnOnAutoTLS(t, tlsClients)
+	// curl HTTPS
+	secretName := routenames.Certificate(objects.Route)
+	rootCAs := createRootCAs(t, clients, objects.Route.Namespace, secretName)
+	httpsClient := createHTTPSClient(t, clients, objects, rootCAs)
+	testingress.RuntimeRequest(t, httpsClient, "https://"+objects.Service.Status.URL.Host)
+}
+
+func TestPerNamespaceCert_localCA(t *testing.T) {
+	clients := e2e.Setup(t)
+	disableNamespaceCertWithWhiteList(t, clients, sets.NewString(test.ServingNamespace))
+
+	cancel := turnOnAutoTLS(t, clients)
 	defer cancel()
 
 	// wait for certificate to be ready
-	waitForCertificateReady(t, tlsClients, routenames.Certificate(objects.Route))
+	certName := waitForNamespaceCertReady(t, clients)
+
+	// Create Knative Service
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "runtime",
+	}
+	test.CleanupOnInterrupt(func() { test.TearDown(clients, names) })
+	objects, err := v1test.CreateServiceReady(t, clients, &names)
+	if err != nil {
+		t.Fatalf("Failed to create initial Service: %v: %v", names.Service, err)
+	}
 
 	// curl HTTPS
-	httpsClient := createHTTPSClient(t, tlsClients, objects, rootCAs)
+	rootCAs := createRootCAs(t, clients, test.ServingNamespace, certName)
+	httpsClient := createHTTPSClient(t, clients, objects, rootCAs)
 	testingress.RuntimeRequest(t, httpsClient, "https://"+objects.Service.Status.URL.Host)
 }
 
-func createHTTPSClient(t *testing.T, tlsClients *autoTLSClients, objects *v1test.ResourceObjects, rootCAs *x509.CertPool) *http.Client {
-	ing, err := tlsClients.clients.NetworkingClient.Ingresses.Get(routenames.Ingress(objects.Route), metav1.GetOptions{})
+func createRootCAs(t *testing.T, clients *test.Clients, ns, secretName string) *x509.CertPool {
+	secret, err := clients.KubeClient.Kube.CoreV1().Secrets(ns).Get(
+		secretName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to get Secret %s: %v", secretName, err)
+	}
+
+	rootCAs, _ := x509.SystemCertPool()
```

don't ignore the error.